crud/select/plan.lua

```diff
@@ -48,49 +48,75 @@ local function get_index_for_condition(space_indexes, space_format, condition)
     end
 end
 
-local function extract_sharding_key_from_scan_value(scan_value, scan_index, sharding_index)
-    if #scan_value < #sharding_index.parts then
+local function extract_sharding_key_from_conditions(conditions, sharding_index, space_indexes, fieldno_map)
```

Comment:
I would add unit tests for `extract_sharding_key_from_conditions()`. I see, integration tests covers function in general, but unit tests are simple to understand and more fast in execution than integration tests. At least case `sharding_index == nil` is not covered in integration tests.

Reply:
Added tests, `sharding_key == nil` is not supported anymore